### PR TITLE
Unpinning CI python versions as no longer needed

### PR DESCRIPTION
**Pull request by jacobcook1995**
# Description

Github actions python versions for 3.9 and 3.10 now default to 3.9.16 and 3.10.9, respectively. This means there's no longer any reason to keep the versions pinned, so I have unpinned them. (They were originally pinned to fix a weird CI edge case where a specific test would fail on macOS for python versions <3.9.16 and <3.10.9)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
         # Pinned to fix problem with macosx and older python versions (change when fixed)
```
Comment (davidorme) [inline .github/workflows/ci.yml:22]: Remove comment.

**Review by davidorme — APPROVED** [review]
Sounds good! Might as well remove out of date comment too
